File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int i = 0, j = 1;
        int currProfit = 0;
        while(j < prices.length) {
            if(prices[i] < prices[j]) {
                currProfit = Math.max(currProfit, prices[j] - prices[i]);
            }
            else {
                i = j;
            }           
            j++;
        }
        return currProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int buy = prices[0];
        int maxProfit = 0;

        for(int day = 1; day < prices.Length; day++) {
            if(buy > prices[day]) {
                buy = prices[day];
            }
            else {
                maxProfit = Math.Max(maxProfit, prices[day] - buy);
            }
        }

        return maxProfit;
    }
}